### ISSUE TITLE
9.4.3 : Mail collector does not collect if a mail contains only the preformatted subject

### ISSUE THREAD

**Issue by tuxxic** [glpi-project/glpi#6442]
**Describe the bug**

When a Requester replies to the mail notification with the right subject (eg: [GLPI #number]) without keeping the magic GLPI line (in French, =-=-=-= Pour répondre par courriel, écrivez au dessus de cette ligne =-=-=-= ), the mail is never collected and triggers an error in case I wish to force collect:

![opera_2019-10-10_11-01-44](https://user-images.githubusercontent.com/4680438/66556127-f9f46d00-eb4f-11e9-88f9-bdcb2403a3e3.png)

**Page(s) URL**

If applicable, page(s) URL where the bug happens.

**To reproduce**

1. Reply to a notification without the magic line but with the right subject
2. Try to collect : the mail is never collected.

**Expected behavior**

I wished at least the mail to be collected as a new ticket, that I could "Merge as followup" if needed.

**Comment by tuxxic** [comment]
Thanks,
It is a collected mail, so of course there is no information. I configured "Behaviours", which may be the problem. I will try making two more instances for testing.

**Comment by cedric-anne** [comment]
Hi,

The error seems to indicate that you are trying to reply on a solved ticket.
Is the behaviour the same if you just reply to a ticket that is not closed not solved ?

Regards

**Comment by tuxxic** [comment]
Hi cedric-anne,

In fact I have a customer e-mail in the collecting folder which has the properties I gave in my description.

Other tickets are collected correctly, and replies are functional.

**Comment by cedric-anne** [comment]
Is this reply related to a ticket having a solution ?

**Comment by tuxxic** [comment]
Yes

**Comment by cedric-anne** [comment]
Looks like GLPI consider content of the followup as empty.

To debug this case, we will need raw email as .eml file.

Regards

**Comment by tuxxic** [comment]
I think you can easily reproduce by getting data in the mail without the magic GLPI lines, and having the [GLPI #number] in the subject.

Unfortunately, I do not have permission to give away the .eml file ...

**Comment by cedric-anne** [comment]
Hi,

Notification emails two special lines (which are translated):
 - header line: `=-=-=-= To answer by email, write above this line =-=-=-=`,
 - footer line: `=_=_=_= To answer by email, write under this line =_=_=_=`.

When a mail is collected:
 - if these 2 lines are present, all the content between these lines is stripped,
 - if only header line is resent, all the content from the header line to the end is stripped,
 - if only footer line is resent, all the content from the begining to the footer line is stripped.

So, basically, if your user is removing the header line and writes nothing above the footer line, resulting content is empty, and email no followup is added.

As explained #6503, we will not fix this behaviour for now, as this may require lot of work and almost no gain for end-user. Indeed, as usable content is empty, we cannot do anything with this mail, so we would just marked it as "ignored" instead of "errored".

Regards

**Comment by RobertoMarcelino** [comment]
I usually say: what the eyes do not see the heart does not feel.

So if the user does not see the lines, he will not be tempted to delete.

I resolved as follows: I edited the files /usr/share/glpi/inc/notificationtargetticket.class.php and /usr/share/glpi/inc/notificationtemplate.class.php.

In the /usr/share/glpi/inc/notificationtargetticket.class.php file, change:

Lines 42 and 43 of:
<code>
const HEADERTAG = '=-=-=-=';
const FOOTERTAG = '=_=_=_=';
</code>

for:
<code>const HEADERTAG = '<span style="font-size:6pt;color: #fff;">=-=-=-=</span>';
const FOOTERTAG = '<span style="font-size:6pt;color: #fff;">=_=_=_=</span>';</code>

Lines 80 and 81 of:
<code>return self::HEADERTAG.' '.__('To answer by email, write above this line').' '.
                self::HEADERTAG;</code>

for:
<code>return self::HEADERTAG.' '.__('').' '.
                self::HEADERTAG;</code>

Lines 96 and 97 of:
<code>return self::FOOTERTAG.' '.__('To answer by email, write under this line').' '.
                self::FOOTERTAG;</code>

for:
<code>return self::FOOTERTAG.' '.__('').' '.
                self::FOOTERTAG;</code>

In the /usr/share/glpi/inc/notificationtemplate.class.php file, change:

Line 256 of:
<code>$footer_string = Html::entity_decode_deep(__('Automatically generated by GLPI'));</code>

for:
<code>$footer_string = Html::entity_decode_deep(__(''));</code>

Lines 309 through 314:
<code><body>\n".(!empty($add_header)?$add_header."\n<br><br>":'').
$template_datas['content_html'].
"<br><br>-- \n<br>".$signature_html.
"<br>$footer_string".
"<br><br>\n".(!empty($add_footer)?$add_footer."\n<br><br>":'').
"\n</body></html>";</code>

for:
<code><body>".(!empty($add_header)?$add_header."":'').
$template_datas['content_html'].
"".$signature_html.
"$footer_string".
"".(!empty($add_footer)?$add_footer."":'').
"</body></html>";</code>

The lines needed for GLPI to function will be there, but in very small size and hidden (blank).
